src/parser/toplevel.lisp

```diff
@@ -2069,8 +2074,32 @@ consume all attributes")))
                  :message "Malformed struct field"
                  :primary-note "expected field type")))
 
+  ;; (name "docstring") missing ty
+  (when (and (cst:consp form)
+             (cst:atom (cst:rest form))
+             (stringp (cst:raw (cst:rest form))))
+    (error 'parse-error
+           :err (coalton-error
+                 :span (cst:source form)
+                 :file file
+                 :message "Malformed struct field"
+                 :primary-note "expected field type after docstring"
+                 :highlight :end)))
+
   ;; (name ty ...)
-  (when (cst:consp (cst:rest (cst:rest form)))
+  (when (and (cst:consp (cst:rest (cst:rest form)))
+             (not (stringp (cst:raw (cst:second form)))))
```

Comment:
You need to check if the third form of the struct field is a list. Calling `cst:raw` on a `cst:cons` will signal an error.